### PR TITLE
Code-review branch

### PR DESCRIPTION
**Pull request by Obaloluwa123**

### CODE REVIEW

**Comment by HanLinAung88** [comment]
It'll be great if you can add a description next time on detailing the changes of this pull request so that the reviewer can get some context into what the change is about

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java [hunk 1]
```diff
@@ -116,4 +130,40 @@ public void onSuggestionClicked(IngredientSearchSuggestion suggestion) {
...
+            ParseQuery<Ingredient> query = ParseQuery.getQuery(Ingredient.class);
+            query.include(Ingredient.USER_KEY);
+            query.findInBackground(new FindCallback<Ingredient>() {
+                @Override
+                public void done(List<Ingredient> ingredients, ParseException e) {
+                    if (e != null) {
+                       Log.e(TAG, "Issue with getting ingredients", e) ;
+                       return;
+                    }
+                    for (Ingredient ingredient: ingredients) {
```
Comment (HanLinAung88) [inline app/src/main/java/com/example/fooding/fragments/PantryFragment.java:162]: What is the purpose of this for loop here?

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java [hunk 2]
```diff
@@ -116,4 +130,40 @@ public void onSuggestionClicked(IngredientSearchSuggestion suggestion) {
...
+        }
+
+        @Override
+        public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
+            selectedSuggestions.remove(viewHolder.getAdapterPosition());
+            suggestionsAdapter.notifyDataSetChanged();
+            suggestionsAdapter.notifyItemRemoved(viewHolder.getAdapterPosition());
+            Toast.makeText(getContext(), "Ingredient deleted", Toast.LENGTH_SHORT).show();
+
+
```
Comment (HanLinAung88) [inline app/src/main/java/com/example/fooding/fragments/PantryFragment.java:147]: nit: Let's remove these empty lines

File: app/src/main/java/com/example/fooding/adapters/SelectedIngredientsAdapter.java [hunk 3]
```diff
@@ -37,15 +40,18 @@ public int getItemCount() {
...
 
     public class ViewHolder extends RecyclerView.ViewHolder {
-        TextView ingredientNameTextView;
+        TextView tvIingredientName;
+
 
         public ViewHolder(@NonNull View itemView) {
             super(itemView);
-            ingredientNameTextView = itemView.findViewById(R.id.ingredientName);
+            tvIingredientName = itemView.findViewById(R.id.ingredientName);
```
Comment (HanLinAung88) [inline app/src/main/java/com/example/fooding/adapters/SelectedIngredientsAdapter.java:48]: Maybe because i am out of context, but what does tVIingredientName refer to? I also see IVIingredientName?

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java (hunk 1 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/fragments/PantryFragment.java:162]: I removed it already, I was just trying to get the ingredients from the API, but i don't need it

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java (hunk 2 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/fragments/PantryFragment.java:147]: I have removed those.

File: app/src/main/java/com/example/fooding/adapters/SelectedIngredientsAdapter.java (hunk 3 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/adapters/SelectedIngredientsAdapter.java:48]: That was the id I gave to the ingredient name and image, but I modified the naming already. i will commit and push the updated code

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java [hunk 4]
```diff
@@ -31,6 +40,8 @@ public class PantryFragment extends Fragment implements SearchView.OnQueryTextLi
     private SearchView recipeSearchView;
     private RecyclerView ingredientsRecyclerView;
     private RecyclerView suggestedRecyclerView;
+    private TextView ingredientName;
+    private TextView ingredientImage;
```
Comment (max-is-hacks) [inline app/src/main/java/com/example/fooding/fragments/PantryFragment.java:44]: A small nit pick. `ingredientName` and `ingredientImage` are `TextView`'s, so perhaps rename them to indicate they are `TextView` objects. 

```suggestion
    private TextView ingredientNameTextView;
    private TextView ingredientImageTextView;
```

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java [hunk 5]
```diff
@@ -51,6 +62,8 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         recipeSearchView = view.findViewById(R.id.ingredientSearchView);
         ingredientsRecyclerView = view.findViewById(R.id.ingredientsRecyclerView);
         suggestedRecyclerView = view.findViewById(R.id.suggestedIngredientsRecyclerView);
+        ingredientName = view.findViewById(R.id.ingredientName);
+        ingredientImage = view.findViewById(R.id.ingredientImage);
```
Comment (max-is-hacks) [inline app/src/main/java/com/example/fooding/fragments/PantryFragment.java:66]: These TextViews are created but never used `PantryFragment`. Maybe this should be removed from this PR and put into a separate PR when the TextViews are used. 

It's a good habit to keep PRs concise and free of unused code.

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java [hunk 6]
```diff
@@ -116,4 +130,40 @@ public void onSuggestionClicked(IngredientSearchSuggestion suggestion) {
...
+            query.findInBackground(new FindCallback<Ingredient>() {
+                @Override
+                public void done(List<Ingredient> ingredients, ParseException e) {
+                    if (e != null) {
+                       Log.e(TAG, "Issue with getting ingredients", e) ;
+                       return;
+                    }
+                    for (Ingredient ingredient: ingredients) {
+
+                    }
```
Comment (max-is-hacks) [inline app/src/main/java/com/example/fooding/fragments/PantryFragment.java:164]: Oops looks like this is an empty loop left in. You could remove it, add the implementation for it, or have the implementation be part of a subsequent PR.

**Review by max-is-hacks — CHANGES_REQUESTED** [review]
Looks like there are some smalls nit pick in the code with empty code implementations  and variable names that could be more clear. 

Keep up the work! : )

File: app/src/main/java/com/example/fooding/adapters/SuggestionsAdapter.java [hunk 7]
```diff
@@ -44,14 +47,19 @@ public interface SuggestionAdapterListener {
 
     public class ViewHolder extends RecyclerView.ViewHolder {
         TextView ingredientNameTextView;
+        ImageView ivIingredientImage;
```
[the remark this answers is not in the document]
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/adapters/SuggestionsAdapter.java:50]: okay, I'll apply that as I write more lines of code

[the remark this answers is not in the document]
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/adapters/SuggestionsAdapter.java:50]: I changed the name already

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java (hunk 4 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/fragments/PantryFragment.java:44]: I have applied the changes. Thank you.

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java (hunk 5 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/fragments/PantryFragment.java:66]: okay, I have removed those lines of code.

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java (hunk 6 above)
Comment (Obaloluwa123) [reply app/src/main/java/com/example/fooding/fragments/PantryFragment.java:164]: Thanks, I removed that already too.

File: app/src/main/java/com/example/fooding/adapters/SelectedIngredientsAdapter.java [hunk 8]
```diff
@@ -52,7 +52,7 @@ public ViewHolder(@NonNull View itemView) {
         }
 
         public void bind(Ingredient suggestion) {
-            tvIingredientName.setText(suggestion.getIngredient());
+            tvIingredientName.setText(suggestion.getIngredientName());
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/adapters/SelectedIngredientsAdapter.java:55]: Please change this back to IngredientNameTextView

File: app/src/main/java/com/example/fooding/adapters/RecipeExploreAdapter.java [hunk 9]
```diff
@@ -1,5 +1,67 @@
...
+import com.bumptech.glide.Glide;
+import com.example.fooding.R;
+import com.example.fooding.models.Ingredients;
+
+import java.util.List;
+
+public class RecipeExploreAdapter extends RecyclerView.Adapter<RecipeExploreAdapter.ViewHolder> {
+
+    Context context;
+    public List<Ingredients> recipeByIngredients;
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/adapters/RecipeExploreAdapter.java:22]: consider renaming to FridgeIngredientList

File: app/src/main/java/com/example/fooding/fragments/InstructionsFragment.java [hunk 10]
```diff
@@ -33,7 +35,12 @@ public static InstructionsFragment newInstance(FoodExtended food) {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.fragment_instructions, container, false);
+        View view = inflater.inflate(R.layout.fragment_instructions, container, false);
+        WebView webView = (WebView) view.findViewById(R.id.instructionsWebview);
+        webView.getSettings().setJavaScriptEnabled(true);
+        webView.setWebViewClient(new WebViewClient());
+        webView.loadUrl("https://fullbellysisters.blogspot.com/");
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/fragments/InstructionsFragment.java:42]: Move the url here to a constant

File: app/src/main/java/com/example/fooding/fragments/PantryFragment.java [hunk 11]
```diff
@@ -184,19 +189,22 @@ private void queryIngredients() {
...
-        query.findInBackground((ingredients, e) -> {
-            // check for errors
-            if (e != null) {
-                Log.e(TAG, "Issue with getting posts", e);
-                return;
-            }
+        query.findInBackground(new FindCallback<Ingredient>() {
+            @Override
+            public void done(List<Ingredient> ingredients, ParseException e) {
+                // check for errors
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/fragments/PantryFragment.java:195]: please add a TODO

File: app/src/main/java/com/example/fooding/fragments/RecipeExploreFragment.java [hunk 12]
```diff
@@ -50,22 +61,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        recipeByIngredients = new ArrayList<>();
+        exploreRecyclerView.setAdapter(recipeAdapter);
+        exploreRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
     }
 
     private void queryIngredients() {
         ParseQuery<Ingredient> query = ParseQuery.getQuery(Ingredient.class);
         query.include(Ingredient.USER_KEY);
-        query.setLimit(20);
+        query.setLimit(120);
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/fragments/RecipeExploreFragment.java:74]: make this a constant and add TODO - stretch goal implement infinite scrolling

File: app/src/main/java/com/example/fooding/fragments/RecipeExploreFragment.java [hunk 13]
```diff
@@ -74,7 +92,7 @@ private void fetchRecipes(List<Ingredient> ingredients) {
         if (ingredients.isEmpty()) return;
         Set<String> uniqueIngredients = new HashSet<>();
         for (Ingredient ingredient : ingredients) {
-            uniqueIngredients.add(ingredient.getIngredient());
+            uniqueIngredients.add(ingredient.getIngredientName());
```
Comment (samkaks) [inline app/src/main/java/com/example/fooding/fragments/RecipeExploreFragment.java:95]: rename to uniqueIngredientSet

**Review by samkaks — CHANGES_REQUESTED** [review]
